kong/pdk/response.lua

```diff
@@ -760,6 +849,196 @@ local function new(self, major_version)
   end
 
 
+  local function get_response_type(content_header)
+    local type = CONTENT_TYPE_JSON
+
+    if content_header ~= nil then
+      local accept_values = split(content_header, ",")
+      local hi_level = 100
+      local hi_quality = 0
+      for _, value in ipairs(accept_values) do
+        local mimetype_values = split(value, ";")
+        local name
+        local level = 1
+        local quality = 1
+        for _, entry in ipairs(mimetype_values) do
+          local m = ngx.re.match(entry, [[^\s*(\S+\/\S+)\s*$]], "ajo")
+          if m then
+            name = m[1]
+          else
+            m = ngx.re.match(entry, [[^\s*q=([0-9]*[\.][0-9]+)\s*$]], "ajoi")
+            if m then
+              quality = tonumber(m[1])
+            else
+              m = ngx.re.match(entry, [[^\s*level=(\d)\s*$]], "ajoi")
+              if m then
+                level = tonumber(m[1])
+              end
+            end
+          end
+        end
+
+        if level <= hi_level and quality > hi_quality then
```

Comment:
I'm a bit confused by the algorithm used here. Perhaps it's because of the names of `hi_level` and `hi_quality` It seems we are looking for the *highest* quality and the *lowest* level? (so it would be `min_level`, `max_quality`)

Apart from the naming, I've been reading [RFC 7231, section 5.3.2](https://tools.ietf.org/html/rfc7231#section-5.3.2) and I have two concerns:

* It seems that this parser won't be able to satisfy requests with media ranges such as `Accept: text/*` (it will find no match on the name, and return the default type, `CONTENT_TYPE_JSON`, instead of html or txt).
* I am also almost certain that the `level` attribute shouldn't be used. See https://stackoverflow.com/a/14527918

Reply:
Also I think this advanced parsing should not be done with Content-Tupe, only with Accept.

Reply:
>Also I think this advanced parsing should not be done with Content-Tupe, only with Accept.

@bungle The `Content-Type` is used to enable the plugins to ignore the `Accept` header, do you think it is not needed?

Reply:
What I mean that `Content-Type` header's content is not to be parsed like `Accept` header content. I mean that:
1. check if there is accept header and parse
2. if not check if there is content-type header and just use that, without parsing it.